File: frontend/src/api/exports.ts
import { API_BASE_URL } from './client';
import { AnswerResponse } from './answers';

export async function exportToMarkdown(answerData: AnswerResponse): Promise<Blob> {
  const response = await fetch(`${API_BASE_URL}/api/v1/exports/markdown`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ answer_data: answerData }),
  });

  if (!response.ok) {
    const errorData = await response.json().catch(() => ({}));
    throw new Error(errorData.detail || 'Export failed');
  }

  return await response.blob();
}

File: backend/app/api/routes/exports.py
from fastapi import APIRouter, Depends, Response
from app.schemas.exports import MarkdownExportRequest
from app.services.exports.markdown_export_service import MarkdownExportService

router = APIRouter()

def get_markdown_export_service() -> MarkdownExportService:
    return MarkdownExportService()

@router.post("/markdown", response_class=Response)
async def export_to_markdown(
    request: MarkdownExportRequest,
    service: MarkdownExportService = Depends(get_markdown_export_service)
):
    """
    Exports a synthesized answer into a downloadable Markdown file.
    """
    try:
        markdown_content = service.generate_markdown(request.answer_data)

        return Response(
            content=markdown_content,
            media_type="text/markdown",
            headers={
                "Content-Disposition": 'attachment; filename="knowledge_map_export.md"'
            }
        )
    except Exception as e:
        from fastapi import HTTPException
        raise HTTPException(status_code=500, detail=f"Export failed: {str(e)}")
